### PR TITLE
Dunify and be compatible with MirageOS 4

### PR DESCRIPTION
**Pull request by dinosaure**

### CODE REVIEW

**Comment by dinosaure** [comment]
I will figure out why `ocaml-ci` fails to compile this version of `Zarith`. At least, this version works for MirageOS 4.

**Comment by samoht** [comment]
This needs a `dune build @fmt --auto` to fix the formatting of the new dune files

**Comment by dinosaure** [comment]
To keep the diff clear, I disabled `ocamlformat` 👍 .

File: .ocamlformat [hunk 1]
```diff
@@ -0,0 +1,2 @@
+version=0.20.1
```
Comment (samoht) [inline .ocamlformat:1]:
```suggestion
```

File: .ocamlformat [hunk 2]
```diff
@@ -0,0 +1 @@
+disable=true
```
Comment (samoht) [inline .ocamlformat:2]:
```suggestion
version=0.20.1
disable=true
```

File: .ocamlformat [hunk 3]
```diff
@@ -0,0 +1,3 @@
+version=0.20.0
```
Comment (samoht) [inline .ocamlformat]:
```suggestion
```

**Comment by samoht** [comment]
The CI issue should be solved by https://github.com/ocurrent/ocaml-ci/pull/403

**Pull request merged**